src/AgentCommerce/SalesChannel/CreateCartRoute.php

```diff
@@ -63,8 +76,44 @@ public function createCart(Request $request, SalesChannelContext $salesChannelCo
         $swCart = $this->cartService->createNew($salesChannelContext->getToken());
         $swCart = $this->cartService->add($swCart, $lineItems, $salesChannelContext);
 
+        $order = $this->orderBuilder->getOrderFromCart($swCart, $salesChannelContext, new RequestDataBag($request->request->all()));
+        $orderId = $request->attributes->get(AbstractPaymentMethodHandler::PAYPAL_PAYMENT_ORDER_ID_INPUT_NAME);
+
+        if ($orderId) {
+            $itemList = $this->itemListProvider->getItemListFromCart($salesChannelContext->getCurrency(), $swCart);
+
+            $purchaseUnit = $this->purchaseUnitProvider->createPurchaseUnit(
+                $swCart->getTransactions()->first()->getAmount(),
+                $swCart->getShippingCosts(),
+                $salesChannelContext->getCustomer(),
+                $itemList,
+                $salesChannelContext->getCurrency(),
+                $salesChannelContext->getContext(),
+                $swCart->getPrice()->hasNetPrices(),
+            );
+
+            $purchaseUnit = $order->getPurchaseUnits()->first();
+
+            $purchaseUnitArray = \json_decode((string) \json_encode($purchaseUnit), true);
+
+            $purchaseUnitPatch = new Patch();
+            $purchaseUnitPatch->assign([
+                'op' => Patch::OPERATION_REPLACE,
+                'path' => '/purchase_units/@reference_id==\'default\'',
+            ]);
+
+            $purchaseUnitPatch->setValue($purchaseUnitArray);
+
+            $this->orderResource->update([$purchaseUnitPatch], $orderId, $salesChannelContext->getSalesChannelId(), PartnerAttributionId::PAYPAL_PPCP);
+            $order = $this->orderResource->get($orderId, $salesChannelContext->getSalesChannelId());
```

Comment:
The order is not needed further down the line, so there is no need to refetch it